### ISSUE TITLE
auto_paging_iter() does not persist params

### ISSUE THREAD

**Issue by mattbreeden** [stripe/stripe-python#221]
From the documentation at https://stripe.com/docs/api/python#auto_pagination:

> To use the auto-pagination feature in Python, simply issue an initial "list" call with the parameters you need, then call auto_paging_iter() on the returned list object to iterate over all objects matching your initial parameters.

With this code example:
```python
customers = stripe.Customer.all(limit=3)
for customer in customers.auto_paging_iter():
  # Do something with customer
```

However using this code the params that are passed in initially do not get saved here: https://github.com/stripe/stripe-python/blob/efe1ecc575ad97d22ad6381ee1f6686f34fc27b1/stripe/resource.py#L391
so the next trip through the params will be empty: https://github.com/stripe/stripe-python/blob/efe1ecc575ad97d22ad6381ee1f6686f34fc27b1/stripe/resource.py#L328

**Comment by brandur** [comment]
Oof, thanks for reporting @mattbreeden! Sorry about the trouble here.

@kyleconroy Any chance that you want to take a shot at this one? Otherwise we can probably put something on the schedule.

**Comment by kyleconroy** [comment]
Yeah, I can take a pass today.

**Comment by brandur** [comment]
@kyleconroy Awesome! Let me know if I can help with a review or something.

**Comment by olivierbellone** [comment]
Submitted a patch here: #222.

**Comment by brandur** [comment]
Released as 1.32.1! Sorry about the trouble here.
